src/reader.rs

```diff
@@ -144,6 +144,160 @@ impl Reader {
+                });
+
+            if remaining_keys.is_empty() {
+                let results = results_converter(results, ttl_now);
+                return Ok(results);
+            }
+        }
+
+        if self.include_memtables(options.durability_filter) {
+            std::iter::once(snapshot.memtable())
+                .chain(snapshot.imm_memtable().iter().map(|imm| imm.table()))
+                .for_each(|memtable| {
+                    for (key, i) in remaining_keys.iter() {
+                        if results[*i].is_none() {
+                            results[*i] = memtable.get(key);
+                        }
+                    }
+                    remaining_keys.retain(|(_, i)| results[*i].is_none());
+                });
+
+            if remaining_keys.is_empty() {
+                let results = results_converter(results, ttl_now);
+                return Ok(results);
+            }
+        }
+
+        // Since the key remains unchanged during the point query, we only need to compute
+        // the hash value once and pass it to the filter to avoid unnecessary hash computation
+        let mut key_hashes = vec![0u64; results.len()];
+        for (key, i) in remaining_keys.iter() {
+            key_hashes[*i] = filter::filter_hash(key);
+        }
+
+        // cache blocks that are being read
+        let sst_iter_options = SstIteratorOptions {
+            cache_blocks: true,
+            eager_spawn: true,
+            ..SstIteratorOptions::default()
+        };
+
+        for sst in &snapshot.core().l0 {
+            for (key, i) in remaining_keys.iter() {
+                let filter_result = self.sst_might_include_key(sst, key, key_hashes[*i]).await?;
+                self.record_filter_result(&filter_result);
+                if matches!(filter_result, RangeNegative) {
+                    break;
+                }
+
+                if filter_result.might_contain_key() {
+                    let mut iter =
+                        SstIterator::for_key(sst, key, self.table_store.clone(), sst_iter_options)
+                            .await?;
+
+                    if let Some(entry) = iter.next_entry().await? {
+                        if entry.key == key {
+                            results[*i] = Some(entry);
+                            continue;
+                        }
+                        if matches!(filter_result, FilterPositive) {
+                            self.db_stats.sst_filter_false_positives.inc();
+                        }
+                    }
+                }
+            }
+            remaining_keys.retain(|(_, i)| results[*i].is_none());
+
+            if remaining_keys.is_empty() {
+                let results = results_converter(results, ttl_now);
+                return Ok(results);
+            }
+        }
+
+        for sr in &snapshot.core().compacted {
+            let mut idx = 0;
+            let mut sr_iter = SortedRunIterator::new_borrowed(
```

Comment:
WDYT about `from_key` with the `remaining_keys[0]` as the key? Since they're sorted, I think that should work, and would bring the SR iterator immediately to the first key.